gdal/doc/source/development/rfc/rfc76_ogrpythondrivers.rst

```diff
@@ -0,0 +1,397 @@
+.. _rfc-76:
+
+================================================================================
+RFC 76: OGR Python drivers
+================================================================================
+
+============== ============================
+Author:        Even Rouault
+Contact:       even.rouault @ spatialys.com
+Started:       2019-Nov-5
+Last updated:  2019-Nov-5
+Status:        Draft
+Target:        GDAL 3.1
+============== ============================
+
+Summary
+-------
+
+This RFC adds the capability to write OGR/vector drivers in Python.
+
+Motivation
+----------
+
+For some use cases that do not require lighting speed, or to deal with very
+niche formats (possibly in house format), it might be faster and more efficient
+to write a vector driver in Python rather than a GDAL C++ driver as currently required,
+or an ad-hoc converter.
+
+.. note::
+
+    QGIS has now a way to create Python-based providers such as
+    in https://github.com/qgis/QGIS/blob/master/tests/src/python/provider_python.py
+    Having a way to do in GDAL itself also allows the rest of GDAL/OGR based
+    tools to use the OGR Python driver.
+
+How does that work ?
+--------------------
+
+Driver registration
++++++++++++++++++++
+
+The driver registration mechanism is extended to look for .py scripts in a
+dedicated directory ({prefix}/lib/gdalplugins/python if hardcoded on compilation
+for Unix builds, or the GDAL_PYTHON_DRIVER_PATH environment variable).
+Those Python script must have in their first lines at least 2 global variables
+set:
+
+- ``DRIVER_NAME = "short_name"``
+- ``DRIVER_SUPPORTED_API_VERSION = 1`` . Currently only 1 supported. If the
+  interface changed in a backward incompatible way, we would increment internally
+  the supported API version number. This item enables us to check if we are able
+  to "safely" load a Python driver. If a Python driver would support several API
+  versions (not clear if that's really possible at that point), it might use an
+  array syntax to indicate that, like ``[1,2]``
+- ``DRIVER_DCAP_VECTOR = "YES"``
+- ``DRIVER_DMD_LONGNAME = "my super plugin"``
+
+Optional metadata such as ``DRIVER_DMD_EXTENSIONS`` or
+``DRIVER_DMD_HELPTOPIC`` can be defined (basically, any driver metadata key string prefixed by
+``DRIVER_``
+
+Those will be parsed in a pure textual way, without invocation of the Python
+interpreter, both for efficiency consideration and also because we want to
+delay the research or launch of the Python interpreter as much as possible
+(the typical use case if GDAL used by QGIS: we want to make sure that QGIS
+has itself started Python, to reuse that Python interpreter)
+
+From the short metadata, the driver registration code can instantiate GDALDriver
+C++ objects. When the Identify() or Open() method is invoked on that object,
+the C++ code will:
+
+* start/find the Python interpreter if not already done
```

Comment:
What does the '"interpreter" mean?
A `PyThreadState` or `Py_NewInterpreter`-made sub-interpreter? 
An installed Python interpreter external to the Python built into GDAL?

IMO, this is a bit unclear.

Reply:
I've tried to make that a bit clearer